### PR TITLE
initial commit of csv provider

### PR DESCRIPTION
**Pull request by AjayTripathy**
Initial commit, with simple unit test for mapping field

TODO: 
- CSV reading unit test
- End-to-end tests
- Redownload  of csv on node added to data/hourly refresh
- Fallback to default pricing if csv is not loaded.

### CODE REVIEW

File: pkg/cloud/csvprovider.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	EndTimestamp      string `csv:"EndTimestamp"`
+	InstanceID        string `csv:"InstanceID"`
+	AssetClass        string `csv:"AssetClass"`
+	InstanceIDField   string `csv:"InstanceIDField"`
+	InstanceType      string `csv:"InstanceType"`
+	MarketPriceHourly string `csv:"MarketPriceHourly"`
+	Version           string `csv:"Version"`
+}
+
+func parseMapField(mf string) {
```
Comment (mbolt35) [inline pkg/cloud/csvprovider.go:36]: No-op intentionally?

File: test/cloud_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	n := &v1.Node{}
+	n.Spec.ProviderID = providerIDWant
+	n.Name = nameWant
+	n.Labels = make(map[string]string)
+	n.Labels["foo"] = labelFooWant
+
+	wantPrice := "0.1337"
+
+	c := &cloud.CSVProvider{
+		CSVLocation: "../configs/pricing_schema.csv",
```
Comment (mbolt35) [inline test/cloud_test.go:57]: Should this file be part of the configs if we're adding it to tests?

File: test/cloud_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	unknownN.Labels = make(map[string]string)
+	unknownN.Labels["foo"] = labelFooWant
+	k2 := c.GetKey(n.Labels, unknownN)
+	resN2, _ := c.NodePricing(k2)
+	if resN2 != nil {
+		t.Errorf("CSV provider should return nil on missing node")
+	}
+	
+	c2 := &cloud.CSVProvider{
+		CSVLocation: "../configs/fake.csv",
```
Comment (mbolt35) [inline test/cloud_test.go:83]: Test file included?

Comment (AjayTripathy) [reply test/cloud_test.go:83]: Intentionally left out.

File: test/cloud_test.go (hunk 2 above)
Comment (AjayTripathy) [reply test/cloud_test.go:57]: Added.

File: pkg/cloud/csvprovider.go (hunk 1 above)
Comment (AjayTripathy) [reply pkg/cloud/csvprovider.go]: Removed...accidentaly left it in and the compiler didn't complain hah.

**Pull request merged**